### ISSUE TITLE
i18n out of the box

### ISSUE THREAD

**Issue by saurabhdaware** [abelljs/abell#17]
If we could have content like this,
```
content/
    |- my-cool-blog/
        |- index.md
        |- index_zh-Hans.md
        |- index_hi.md
```

so if a user prefers Mandarin, he will get content from index_zh-Hans.md (`zh-CN` is the code of Chinese Simplified)

**Comment by Judionit** [comment]
Guess this is a great solution, but we will many files with that

**Comment by saurabhdaware** [comment]
yeah fair point, Have you seen any project handling i18n? I am a little clueless about this right now. Are there suppose to be two different websites for two different languages? or just the same website where you switch content with JavaScript?

**Comment by Judionit** [comment]
No, no, same website but separating translations into multiple files and load them on demand

**Comment by saurabhdaware** [comment]
oh yes makes sense

**Comment by Judionit** [comment]
oh yeah! like what you show on the top,
but need to add the code for each language in the documentation

**Comment by kasipavankumar** [comment]
There are two possibilities we can think of.

1. The site can have subdomains based on the language. For instance, the React Documentation's Hindi variant has [hi.reactjs.org](https://hi.reactjs.org) or Spanish variant has [es.reactjs.org](https://es.reactjs.org).
2. When user requests for a translated page, it can routed to respective language sub pages as can be seen at [overreacted.io](https://overreacted.io). For example, a post has URL of `https://overreacted.io/goodbye-clean-code` & Spanish version of the post has URL of `https://overreacted.io/es/goodbye-clean-code`. This logic is probably handled by a Gatsby plugin named `gatsby-plugin-i18n`.

<hr />

References
* [Repository of overreacted.io](https://github.com/gaearon/overreacted.io)
* [Repository of gatsby-plugin-i18n](https://github.com/angeloocana/gatsby-plugin-i18n)

**Comment by saurabhdaware** [comment]
Yes 2nd method seems like a better option. Thank you for the help @kasipavankumar :D
